numerai_tools/submissions.py

```diff
@@ -16,6 +19,105 @@
 SIGNALS_ALLOWED_PRED_COLS = ["prediction", "signal"]
 
 
+def validate_headers(
+    expected_id_cols: List[str], expected_pred_cols: List[str], submission: pd.DataFrame
+) -> Tuple[str, str]:
+    """Validate the given submission has the right headers.
+
+    Arguments:
+        submission -- pandas DataFrame of the submission
+
+    Return Tuple[str, str]:
+        - string name of the id column
+        - string name of the prediction column
+    """
+    expected_headers = [
+        [ticker_col, signal_col]
+        for ticker_col in expected_id_cols
+        for signal_col in expected_pred_cols
+    ]
+    columns = submission.columns
+    valid_headers = list(columns) in expected_headers
+    assert (
+        valid_headers
+    ), f"headers must be one of {expected_id_cols} and one of {expected_pred_cols}"
+    return columns[0], columns[1]
+
+
+def validate_headers_numerai(submission: pd.DataFrame) -> Tuple[str, str]:
+    return validate_headers(
+        NUMERAI_ALLOWED_ID_COLS, NUMERAI_ALLOWED_PRED_COLS, submission
+    )
```

Comment:
This is out of scope for this change. Once I've incorporated these open-sourced validation functions back into the submission validators it will be easier to abstract that code for public consumption. Until then, we are only adding these functions.